Dockerfile

```diff
@@ -12,13 +12,74 @@
 # limitations under the License.
 #
 
-FROM tomcat:9-jdk8-openjdk
+FROM openjdk:8-jre-alpine
 
 LABEL \
     vendor="ABSA" \
     copyright="2020 ABSA Group Limited" \
     license="Apache License, version 2.0" \
     name="Hyperdrive Workflow Manager"
 
+# SET ENVIRONMENT VARIABLES
+ENV TOMCAT_MAJOR=9 \
+    TOMCAT_VERSION=9.0.37 \
+    TOMCAT_HOME=/opt/tomcat \
+    CATALINA_HOME=/opt/tomcat \
+    JAVA_HOME=/usr/lib/jvm/java-1.8-openjdk/jre \
+    HADOOP_HOME=/opt/hadoop \
+    HADOOP_CONF_DIR=/opt/hadoop/etc/hadoop \
+    SPARK_HOME=/opt/spark \
+    SPARK_CONF_DIR=/opt/spark/conf \
+    KRB_HOME=/etc/
+
+ENV PATH $CATALINA_HOME/bin:$HADOOP_HOME/bin:$HADOOP_HOME/sbin:$SPARK_HOME/bin:$PATH
+
+EXPOSE 8080
+
 ARG WAR_FILE
-COPY ${WAR_FILE} /usr/local/tomcat/webapps/hyperdrive_trigger.war
+
+# TOMCAT
+RUN apk upgrade --update && \
+    apk add --update curl && \
+    apk add bash && \
+    apk add --no-cache krb5-pkinit krb5-dev krb5 && \
+    curl -jksSL -o /tmp/apache-tomcat.tar.gz http://archive.apache.org/dist/tomcat/tomcat-${TOMCAT_MAJOR}/v${TOMCAT_VERSION}/bin/apache-tomcat-${TOMCAT_VERSION}.tar.gz && \
+    gunzip /tmp/apache-tomcat.tar.gz && \
+    tar -C /opt -xf /tmp/apache-tomcat.tar && \
+    ln -s /opt/apache-tomcat-${TOMCAT_VERSION} ${TOMCAT_HOME} && \
+    rm -rf ${TOMCAT_HOME}/webapps/* && \
+    apk del curl && \
+    rm -rf /tmp/* /var/cache/apk/*
+
+# USE ROOT USER
+USER 0
+
+# COPY SPARK AND HADOOP BINARIES CPT
+COPY --chown=0:0 install/hadoop-3.2.1 ${HADOOP_HOME}
+COPY --chown=0:0 install/spark-3.0.1-bin-hadoop3.2 ${SPARK_HOME}
+
+# REMOVE DEFAULT CONFIG FILES
+RUN rm -rf /etc/krb5.conf && \
+    rm -rf ${HADOOP_CONF_DIR}/* && \
+    rm -rf ${SPARK_CONF_DIR}/*
+
+# COPY REGION CONFIG FILES
+COPY install/krb/krb5.conf ${KRB_HOME}
+COPY install/hadoop/conf-ire/ ${HADOOP_CONF_DIR}
+COPY install/spark/conf-ire/ ${SPARK_CONF_DIR}
+
```

Comment:
If I understand this correctly, you assume the files in the `install` folder to exist locally. And obviously you can't upload these files into this public repo. In that case, I think it doesn't make a lot of sense to keep this dockerfile here, better to move it to a private repo. The only dependency of this Dockerfile on this repo is the war file. But that's only an argument anyway, so the Dockerfile can live in another repo as well

Reply:
I think you do not need a private repo. Just pass these as ENV variables and make them a path to a mounted volume and have it documented that this is expected. It then does not have to be here or in the image and is passed at runtime not at build time making the image generally usable. On our AWS you can have long-living EFS with the files.

If you hardcode these paths like this, one image can be used per environment and you would have to build a new one each time.

Reply:
I think, ultimately it depends on DevOps what works best for them. But to me, it makes no sense to have a Dockerfile that doesn't work out of the box. I would prefer a private repo that is completely self-contained.

Reply:
My point of view, and I am quite vocal about this, is that an image that you upload needs to be general and "lean". It is also a best practice of Docker to not store application data/specific data in the image but on the volume. A lot of Docker images expect you to pass env variables and/or volumes to them on run time.

Also, your Dockerfile won't work out of the box even if this is a private repo. It will work for one environment only and if everything is not covered by proper DNS names, then the configs will lose their value after each restart. 

Happy to talk more offline if needed.